server/odc-service/src/main/java/com/oceanbase/odc/service/resource/k8s/status/K8sConfigMapStatusDfa.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.resource.k8s.status;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import com.oceanbase.odc.common.dfa.AbstractDfa;
+import com.oceanbase.odc.common.dfa.DfaStateTransfer;
+import com.oceanbase.odc.service.resource.ResourceState;
+import com.oceanbase.odc.service.resource.k8s.model.K8sConfigMap;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link K8sConfigMapStatusDfa}
+ *
+ * @author yh263208
+ * @date 2024-09-12 16:25
+ * @since ODC_release_4.3.2
+ */
+@Slf4j
+public class K8sConfigMapStatusDfa extends AbstractDfa<ResourceState, K8sConfigMap> {
+
+    public static K8sConfigMapStatusDfa getInstance() {
+        List<DfaStateTransfer<ResourceState, K8sConfigMap>> transfers = new ArrayList<>();
```

Comment:
If DFA is a static method with static matchers, why each time a new instance will created.

Reply:
because the dfa object is a stateful object